lib/stripe.ex

```diff
@@ -131,6 +123,14 @@ defmodule Stripe do
     """
   end
 
+  defmodule OAuthAPIError do
+    defexception [:message, :error, :state, :status_code]
+
+    def exception(opts) do
+      struct(opts)
```

Comment:
What do you think about having this be an `%OAuthAPIError{}` rather than a normal struct?